src/app/static/src/tests/integration/genericChart/genericChart_jsonata.itest.ts

```diff
@@ -0,0 +1,248 @@
+            "scrollZoom": false,
+            "modeBarButtonsToRemove": ["zoom2d", "pan2d", "select2d", "lasso2d", "autoScale2d", "resetScale2d", "zoomIn2d", "zoomOut2d"]
+        });
+    });
+
+    it("evaluates top level layout properties as expected", () => {
+        const layout = inputTimeSeriesJsonataResult.layout;
+        expect(Object.keys(layout)).toStrictEqual([
+            "margin", "dragmode", "grid", "annotations",
+            "yaxis1", "xaxis1", "yaxis2", "xaxis2", "yaxis3", "xaxis3"
+        ]);
+        expect(layout.margin).toStrictEqual({"t": 32});
+        expect(layout.dragmode).toBe(false);
+        expect(layout.grid).toStrictEqual({"columns": 2, "rows": 2, "pattern": "independent"});
+    });
+
+    it("evaluates layout annotations as expected", () => {
+        const layout = inputTimeSeriesJsonataResult.layout;
+        expect(JSON.stringify(layout.annotations)).toBe(JSON.stringify([
+            {
+                "text": "Chitipa (MWI_4_1_demo)",
+                "textfont": {},
+                "showarrow": false,
+                "x": 0.5,
+                "xanchor": "middle",
+                "xref": "x1 domain",
+                "y": 1.1,
+                "yanchor": "middle",
+                "yref": "y1 domain"
+            },
+            {
+                "text": "Karonga (MWI_4_2_demo)",
+                "textfont": {},
+                "showarrow": false,
+                "x": 0.5,
+                "xanchor": "middle",
+                "xref": "x2 domain",
+                "y": 1.1,
+                "yanchor": "middle",
+                "yref": "y2 domain"
+            },
+            {
+                "text": "Rumphi (MWI_4_3_demo)",
+                "textfont": {},
+                "showarrow": false,
+                "x": 0.5,
+                "xanchor": "middle",
+                "xref": "x3 domain",
+                "y": 1.1,
+                "yanchor": "middle",
+                "yref": "y3 domain"
+            }
+        ]));
+    });
+
+    it("evaluates layout axes as expected", () => {
+        const layout = inputTimeSeriesJsonataResult.layout;
+
+        const expectedXAxis = {
+            "zeroline": false,
+            "tickvals": ["2011 Q4", "2012 Q4"],
+            "tickfont": {"color": "grey"}
+        };
+
+        expect(layout.xaxis1).toStrictEqual(expectedXAxis);
+        expect(layout.xaxis2).toStrictEqual(expectedXAxis);
+        expect(layout.xaxis3).toStrictEqual(expectedXAxis);
+
+        const expectYAxis = (domainStart: number, domainEnd: number, actual: any) => {
+            expect(Object.keys(actual)).toStrictEqual(["rangemode", "zeroline", "tickformat", "tickfont", "domain"]);
+            expect(actual.rangemode).toBe("tozero");
+            expect(actual.zeroline).toBe(false);
+            expect(actual.tickformat).toBe(".0f");
+            expect(actual.tickfont).toStrictEqual({"color": "grey"});
+            expect(actual.domain[0]).toBeCloseTo(domainStart, 8);
```

Comment:
The floating point values calculated for domain can be a little off (e.g. 0.199999999996) hence using `toBeCloseTo` on these values.